### PR TITLE
Restore backup into azure postgres instance

### PR DESCRIPTION
**Pull request by prabhakarreddy1234**
Create Makerule and corresponding scripts to support restoring a backup file into Azure postgres.

[x] Rebase to main
[x] Attach to trello

### CODE REVIEW

**Comment by prabhakarreddy1234** [comment]
[Postgres backup restore](https://trello.com/c/UqLBetUY/666-postgres-backup-restore)

File: Makefile [hunk 1]
```diff
@@ -15,6 +14,8 @@ dev:
 	$(eval RESOURCE_NAME_PREFIX=s165d01)
 	$(eval ENV_SHORT=dv)
 	$(eval ENV_TAG=dev)
+	$(eval AZURE_BACKUP_STORAGE_ACCOUNT_NAME=s165d01getaniddbbackupdv)
```
Comment (merrilmathew) [inline Makefile:17]: do we need to back up dev database?

File: bin/restore-db [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+if [[ -z "${POSTGRES_PASSWORD}" ]]; then
+  echo "POSTGRES_PASSWORD environment variable not set"
+  exit 1
+fi
+
+if [[ ! -f "${BACKUP_FILENAME}" ]]; then
+  echo "${BACKUP_FILENAME} does not exist."
+  exit 1
+else
+  export PGPASSWORD=${POSTGRES_PASSWORD}
```
Comment (saliceti) [inline bin/restore-db:39]: This may not have to be in the `if` block

Comment (prabhakarreddy1234) [reply bin/restore-db:40]: moved it out of `if` block.

File: Makefile (hunk 1 above)
Comment (prabhakarreddy1234) [reply Makefile]: Kept it just for local testing in `dev` when needed. But we can add it later when needed. so removed it.

File: Makefile [hunk 3]
```diff
@@ -66,6 +71,13 @@ validate-keyvault-secret: read-keyvault-config install-fetch-config set-azure-ac
 	bin/fetch_config.rb -s azure-key-vault-secret:${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} -d quiet \
 		&& echo Data in ${KEY_VAULT_NAME}/${KEY_VAULT_SECRET_NAME} looks valid
 
+restore-data-from-backup: read-keyvault-config read-deployment-config # make production restore-data-from-backup CONFIRM_RESTORE=YES BACKUP_FILENAME="get-an-identity-production-pg-svc-2022-07-06-01"
+	@if [[ "$(CONFIRM_RESTORE)" != YES ]]; then echo "Please enter "CONFIRM_RESTORE=YES" to run workflow"; exit 1; fi
+	$(eval export AZURE_BACKUP_STORAGE_ACCOUNT_NAME=$(AZURE_BACKUP_STORAGE_ACCOUNT_NAME))
+	$(if $(BACKUP_FILENAME), , $(error can only run with BACKUP_FILENAME, eg BACKUP_FILENAME="get-an-identity-production-pg-svc-2022-04-28-01"))
+#bin/download-db-backup ${AZURE_BACKUP_STORAGE_ACCOUNT_NAME} ${AZURE_BACKUP_STORAGE_CONTAINER_NAME} ${BACKUP_FILENAME}.sql.gz
```
Comment (saliceti) [inline Makefile:78]: Can the comment be removed now?

File: bin/restore-db [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  echo "POSTGRES_PASSWORD environment variable not set"
+  exit 1
+fi
+
+if [[ ! -f "${BACKUP_FILENAME}" ]]; then
+  echo "${BACKUP_FILENAME} does not exist."
+  exit 1
+fi
+
+  export PGPASSWORD=${POSTGRES_PASSWORD}
```
Comment (saliceti) [inline bin/restore-db:40]: Please align the block on the left

**Review by saliceti — APPROVED** [review]

**Pull request merged**